asserts/checkers.go

```diff
@@ -38,12 +39,18 @@ func checkMandatory(headers map[string]string, name string) (string, error) {
 	return value, nil
 }
 
-func checkAssertType(assertType AssertionType) (*assertionTypeRegistration, error) {
-	reg := typeRegistry[assertType]
-	if reg == nil {
-		return nil, fmt.Errorf("unknown assertion type: %v", assertType)
+var errInvalidAssertionType = errors.New("invalid assertion type")
+
+func checkAssertType(assertType *AssertionType) error {
+	if assertType == nil {
+		return errInvalidAssertionType
+	}
+	sanity := typeRegistry[assertType.Name]
+	// not the expected registered value
+	if sanity != assertType {
+		return errInvalidAssertionType
```

Comment:
This is helpful, but the error message will be awkward as it's reporting it doesn't know a known type. Would be worth saying what's actually going wrong. E.g. `"internal error: known assertion type %q has incorrect address".